rpcs3/Emu/RSX/Overlays/HomeMenu/overlay_home_menu_components.h

```diff
@@ -0,0 +1,209 @@
+#pragma once
+
+#include "Emu/RSX/Overlays/overlays.h"
+#include "Utilities/Config.h"
+
+namespace rsx
+{
+	namespace overlays
+	{
+		static constexpr u16 menu_entry_height = 40;
+		static constexpr u16 menu_entry_margin = 20;
+		static constexpr u16 available_side_width = (overlay::virtual_width - 6 * menu_entry_margin) / 2;
+		static constexpr u16 element_height = 25;
+
+		enum class page_navigation
+		{
+			stay,
+			back,
+			next,
+			exit
+		};
+
+		struct home_menu_entry : horizontal_layout
+		{
+		public:
+			home_menu_entry(const std::string& text);
+		};
+
+		template <typename T, typename C>
+		struct home_menu_setting : horizontal_layout
+		{
+		public:
+			home_menu_setting(C* setting, const std::string& text) : m_setting(setting)
+			{
+				std::unique_ptr<overlay_element> layout  = std::make_unique<vertical_layout>();
+				std::unique_ptr<overlay_element> padding = std::make_unique<spacer>();
```

Comment:
The design seems to be repeated a lot.
The reason I'm complaining is because using unique_ptr has the side-effect of making the items uncopiable where it wouldn't normally matter.

Reply:
add_element takes a unique_ptr and moves it